node/handlers/catalog.ts

```diff
@@ -25,14 +25,19 @@ const HOP_BY_HOP_HEADERS = [
 const isHopByHopHeader = (header: string) => HOP_BY_HOP_HEADERS.includes(header.toLowerCase())
 
 export async function catalog(ctx: Context) {
-  const {vtex: {account, authToken, operationId, production, route, segmentToken, sessionToken}, query, method} = ctx
+  const { vtex: { account, authToken, operationId, production, route, segmentToken, sessionToken }, query, method } = ctx
   let VtexIdclientAutCookie: string | undefined
   const path = route.params.path as string
 
   if (sessionToken) {
     const { session } = ctx.clients
     const sessionPayload = await session.getSession(sessionToken, ['*'])
-    VtexIdclientAutCookie = ramdaPath(['sessionData', 'namespaces', 'cookie', `VtexIdclientAutCookie_${account}`, 'value'], sessionPayload)
+
+    const isImpersonated = !!sessionPayload?.sessionData?.namespaces?.impersonate?.storeUserId?.value
+
+    VtexIdclientAutCookie = isImpersonated 
+      ? sessionPayload?.sessionData?.namespaces?.cookie?.VtexIdclientAutCookie?.value 
+      : sessionPayload?.sessionData?.namespaces?.cookie?.[`VtexIdclientAutCookie_${account}`]?.value
```

Comment:
It could be cleaner:

```ts
const VtexIdCookieName = isImpersonated ? 'VtexIdclientAutCookie' : `VtexIdclientAutCookie_${account}`
const VtexIdclientAutCookie = sessionPayload?.sessionData?.namespaces?.cookie?.[VtexIdCookieName]?.value
```